src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -8831,7 +8892,7 @@ namespace WaterThermalTanks {
         {
             auto const SELECT_CASE_var(WaterHeaterDesuperheater(DesuperheaterNum).TankTypeNum);
 
-            if (SELECT_CASE_var == MixedWaterHeater) {
+            if (SELECT_CASE_var == MixedWaterHeater||SELECT_CASE_var == StratifiedWaterHeater) {
```

Comment:
This SELECT_CASE_var stuff can be cleaned out and made into a plain IF or SWITCH block.  This is just a carryover from the Fortran to C++ translation.

Reply:
Replaced it with a plain IF block